Some improvements and updates

timhovius opened the pull request:
It's working again now

nickpeirson commented:
Thanks for the PR. The guzzle update looks good.

I'd like to keep the debugging in that was removed in 232279e. I'm happy for this to move to something more sophisticated, e.g. PSR-3, but I don't want it removed without a replacement.

I'm happy with the original request naming and use of fluent interface so these commits need dropping from this branch before I merge (c5fe482, 6e6b032). Adding `Request` to the class names is redundant as they're already in the `Request` namespace. Calling code can alias them to improve clarity in a particularly confusing context, e.g. `use Nickpeirson\Evohome\Request\Gateway as GatewayRequest`.

As the setters added in 4d2f005 aren't yet implemented can you move this commit to a separate branch for merging when they are implemented please.

timhovius commented:
I changed a lot in the code. See the comments below. 

> I'd like to keep the debugging in that was removed in 232279e. I'm happy for this to move to something more sophisticated, e.g. PSR-3, but I don't want it removed without a replacement.

See commit d5105f4

> I'm happy with the original request naming and use of fluent interface so these commits need dropping from this branch before I merge (c5fe482, 6e6b032). Adding Request to the class names is redundant as they're already in the Request namespace. Calling code can alias them to improve clarity in a particularly confusing context, e.g. use Nickpeirson\Evohome\Request\Gateway as GatewayRequest.

Removed, it's now the same as before

> As the setters added in 4d2f005 aren't yet implemented can you move this commit to a separate branch for merging when they are implemented please.

Moved to a new branche named [develop](https://github.com/timhovius/evohome-php/tree/develop)

nickpeirson commented:
Thanks for making those changes :+1:

Everything looks good so I'll get it merged down.
